### PR TITLE
xdg-desktop fuzz case

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
@benzea ptal.

File: src/xdg-autostart-generator/xdg-autostart-service.c [hunk 1]
```diff
@@ -182,6 +182,37 @@ static int xdg_config_parse_string(
         return 0;
 }
 
+static int strv_strdnup_unescape_and_push(
```
Comment (yuwata) [inline src/xdg-autostart-generator/xdg-autostart-service.c]: Maybe `strdnup` -> `strndup`?

Comment (keszybz) [reply src/xdg-autostart-generator/xdg-autostart-service.c]: Fixed.

**Comment by yuwata** [comment]
LGTM.

File: src/xdg-autostart-generator/xdg-autostart-service.c [hunk 2]
```diff
@@ -270,14 +273,12 @@ static int xdg_config_parse_strv(
...
 
-        /* Any trailing entry should be ignored if it is empty. */
-        if (end > start) {
-                r = strv_strndup_unescape_and_push(unit, filename, line,
-                                                   &sv, &n_allocated, &n,
-                                                   start, end);
-                if (r < 0)
-                        return r;
-        }
+        /* Handle the trainling entry after the last separator */
```
Comment (benzea) [inline src/xdg-autostart-generator/xdg-autostart-service.c]: trailing, instead of trainling :)

Comment (keszybz) [reply src/xdg-autostart-generator/xdg-autostart-service.c]: Fixed.

**Comment by keszybz** [comment]
bionic-i386 timed out.

**Pull request merged**